### ISSUE TITLE
Extension crash while return to session

### ISSUE THREAD

**Issue by galen1423** [aunetx/blur-my-shell#136]
Hi,

I am using **Ubuntu 21.10** and after return in active session, **Blur My Shell** has fallen. I reload **G.S** with **Alt+F2 r** and extension working new.

This elements are just after return in session:
```
[Blur my Shell] disabling extension...
[Blur my Shell] removing blur from top panel
[Blur my Shell] removing blur from dash
[Blur my Shell] removing blur from dashes
[Blur my Shell] removing blur from overview
[Blur my Shell] removing blur from lockscreen
[Blur my Shell] removing blur from appfolders
[Blur my Shell] removing blur from window list
[Blur my Shell] extension disabled.
[Blur my Shell] enabling extension...
[Blur my Shell] blurring top panel
[Blur my Shell] blurring dash
[Blur my Shell] searching for dash
[Blur my Shell] dash to dock found, blurring it
JS ERROR: Extension blur-my-shell@aunetx: TypeError: actor is null
connect@/home/galen/.local/share/gnome-shell/extensions/blur-my-shell@aunetx/connections.js:33:18
blur_dash_from@/home/galen/.local/share/gnome-shell/extensions/blur-my-shell@aunetx/dash_to_dock.js:145:26
try_blur@/home/galen/.local/share/gnome-shell/extensions/blur-my-shell@aunetx/dash_to_dock.js:85:35
blur_existing_dashes/<@/home/galen/.local/share/gnome-shell/extensions/blur-my-shell@aunetx/dash_to_dock.js:74:18
blur_existing_dashes@/home/galen/.local/share/gnome-shell/extensions/blur-my-shell@aunetx/dash_to_dock.js:73:37
enable@/home/galen/.local/share/gnome-shell/extensions/blur-my-shell@aunetx/dash_to_dock.js:66:14
enable@/home/galen/.local/share/gnome-shell/extensions/blur-my-shell@aunetx/extension.js:49:37
```

Thanks for your work and I hope I can use it for a long time to come.

**Comment by aunetx** [comment]
Hello, thanks for the issue!
Could you test installing for the branch `tryfix_136`? Tell me if you need help installing from github, and if it works!

I tried to wait some little before the extension tries to use the `connect` function on the null actor, and if it still not appears to just not crash. I hope waiting is actually enough, because I would not like having no callbacks here (it would mean that changing the dock's size would not trigger a blur change, which is not wanted at all)

**Comment by galen1423** [comment]
Hi,

I have try fix136 and error is now : `dash_icons_container is null`

**Comment by aunetx** [comment]
Sorry, did not pass the context to the callback function... might work now

**Comment by galen1423** [comment]
136 --> dash_icons_container is null
137 --> actor is null

**Comment by marcelovbcfilho** [comment]
Hello, don't know if this is useful now, **but** tryfix_136, fixed an issue that I was having that even after boot the extensions wasn't working and now is working.

**Comment by Cloud-Kid** [comment]
> Hello, don't know if this is useful now, **but** tryfix_136, fixed an issue that I was having that even after boot the extensions wasn't working and now is working.

Hey, I took the time to try the [tryfix_136 ](https://github.com/aunetx/blur-my-shell/issues/136) but I still have another error, this is a dash to dock conflict, when I disable the dash to dock extension, blur my shell works with no problems. I'll try to investigate more.

`gnome-shell[24391]: JS ERROR: Extension blur-my-shell@aunetx: TypeError: dash_icons_container is null
                                             blur_dash_from@/home/cloudkid/.local/share/gnome-shell/extensions/blur-my-shell@aunetx/dash_to_dock.js:178:17
                                             try_blur@/home/cloudkid/.local/share/gnome-shell/extensions/blur-my-shell@aunetx/dash_to_dock.js:85:35
                                             blur_existing_dashes/<@/home/cloudkid/.local/share/gnome-shell/extensions/blur-my-shell@aunetx/dash_to_dock.js:74:18
                                             blur_existing_dashes@/home/cloudkid/.local/share/gnome-shell/extensions/blur-my-shell@aunetx/dash_to_dock.js:73:37
                                             enable@/home/cloudkid/.local/share/gnome-shell/extensions/blur-my-shell@aunetx/dash_to_dock.js:66:14
                                             enable@/home/cloudkid/.local/share/gnome-shell/extensions/blur-my-shell@aunetx/extension.js:49:37
                                             connect@/home/cloudkid/.local/share/gnome-shell/extensions/blur-my-shell@aunetx/connections.js:33:18
                                             notify_connect@/home/cloudkid/.local/share/gnome-shell/extensions/blur-my-shell@aunetx/dash_to_dock.js:146:30
                                             blur_dash_from/<@/home/cloudkid/.local/share/gnome-shell/extensions/blur-my-shell@aunetx/dash_to_dock.js:157:41
                                             setTimeout/<@/home/cloudkid/.local/share/gnome-shell/extensions/blur-my-shell@aunetx/utilities.js:10:13
`

**Comment by aunetx** [comment]
I'm sorry everyone I have been really very busy, but I should be able to repair some things during the following days :)

I will look into this tonight if possible! Sorry again :/

**Comment by Cloud-Kid** [comment]
> I'm sorry everyone I have been really very busy, but I should be able to repair some things during the following days :)
> 
> I will look into this tonight if possible! Sorry again :/

No problem, don't worry I'm happy you can maintain the project :) 

It's confirmed on my side it's a conflict with dash to dock, [I tried to disable blur my shell on the dash](https://github.com/aunetx/blur-my-shell/issues/154#issuecomment-990138264) and it works, no more error when waking my PC from lock screen.

**Comment by aunetx** [comment]
This should be fixed by #163 once it is ready

**Comment by aunetx** [comment]
This is normally fixed now, you can test by installing from the `master` branch on github!
